test/YieldDirector.test.js

```diff
@@ -0,0 +1,414 @@
+    // TODO verify this works
+    const calcIndex = (principal, rate, epochs) => principal * (1 + rate) ** epochs;
+
+    // TODO needs cleanup. use Bignumber.
+    // Mine block and rebase. Returns the new index.
+    const triggerRebase = async () => {
+        mineBlock();
+        await staking.rebase();
+
+        return await sOhm.index();
+    }
+
+    before(async () => {
+        [deployer, alice, bob, carol] = await ethers.getSigners();
+        
+        //owner = await ethers.getSigner("0x763a641383007870ae96067818f1649e5586f6de")
+
+        //erc20Factory = await ethers.getContractFactory('MockERC20');
+        // TODO use dai as erc20 for now
+        erc20Factory = await ethers.getContractFactory('DAI');
+
+        stakingFactory = await ethers.getContractFactory('OlympusStaking');
+        ohmFactory = await ethers.getContractFactory('OlympusERC20Token');
+        sOhmFactory = await ethers.getContractFactory('sOlympus');
+        gOhmFactory = await ethers.getContractFactory('gOHM');
+        warmupFactory = await ethers.getContractFactory('StakingWarmup');
+        treasuryFactory = await ethers.getContractFactory('OlympusTreasury');
+        distributorFactory = await ethers.getContractFactory('Distributor');
+        tycheFactory = await ethers.getContractFactory('YieldDirector');
+
+    })
+
+    beforeEach(async () => {
+        //dai = await smock.fake(erc20Factory);
+        //lpToken = await smock.fake(erc20Factory);
+        dai = await erc20Factory.deploy(0);
+        lpToken = await erc20Factory.deploy(0);
+
+        // TODO use promise.all
+        ohm = await ohmFactory.deploy();
+        sOhm = await sOhmFactory.deploy();
+        staking = await stakingFactory.deploy(ohm.address, sOhm.address, "10", "1", "9");
+        gOhm = await gOhmFactory.deploy(sOhm.address, staking.address);
+        treasury = await treasuryFactory.deploy(ohm.address, "0");
+        distributor = await distributorFactory.deploy(treasury.address, ohm.address, "10", "1");
+        warmup = await warmupFactory.deploy(staking.address, sOhm.address);
+        tyche = await tycheFactory.deploy(sOhm.address);
+
+        // Setup for each component
+
+        // Needed for treasury deposit
+        await dai.mint(deployer.address, initialMint);
+        await dai.approve(treasury.address, LARGE_APPROVAL);
+        await sOhm.setgOHM(gOhm.address);
+
+        // Needed to spend deployer's OHM
+        await ohm.approve(staking.address, LARGE_APPROVAL);
+
+        // To get past OHM contract guards
+        await ohm.setVault(treasury.address)
+
+        // Set index to 10
+        await sOhm.setIndex("10000000000");
+        // Initialization for sOHM contract.
+        await sOhm.initialize(staking.address);
+
+        // Set treasury, distributor, warmup, and locker for staking contract
+        await staking.setContract("0", distributor.address);
+        await staking.setContract("1", warmup.address);
```

Comment:
randomly stumbled upon this. `setContract`, according to the code, is expecting to receive the distributor (0), and gOHM (1), but a warmup contract is being sent here?

possibly inconsequencial for this spec, but I'd like to understand. maybe I'm missing something obvious

Reply:
I copied this from the old deploy script from v1.1 and since this was for tests, it didn't cause any issues. But good catch, will change this when i circle back to Tyche.